required/latex-lab/latex-lab-footnotes.dtx

```diff
@@ -0,0 +1,1731 @@
+%    texts is influenced by the setting of the dimen parameter
+%    \cs{footnotemargin}. By default its value is 1.8em in the current
+%    text font (or \texttt{-}\cs{maxdimen} when the para option is
+%    chosen). The following rules apply:
+%    \begin{itemize}
+%    \item
+%
+%      If it has the value \texttt{-}\cs{maxdimen} then the mark is
+%      generated by \cs{@makefnmark}.
+%
+%    \item
+%
+%      Otherwise, if the value is
+%      negative then the mark is placed into an \cs{llap} left aligned
+%      in a box of size \texttt{-}\cs{footnotemargin}.
+%
+%    \item
+%
+%      If the value is zero an \cs{llap} is used without an inner box.
+%
+%    \item
+%
+%      If the value is greater zero (but less than \cs{maxdimen}) the
+%      mark is placed right aligned into a box of size
+%      \cs{footnotemargin}.
+%
+%    \item
+%
+%      The value \cs{maxdimen} is used as a marker to indicate that
+%      no value was given and that the default should be used,
+%      i.e. 1.8em or \texttt{-}\cs{maxdimen} depending on the chosen
+%      option.
+%    \end{itemize}
+%
+%
+% \section{Tagging support}
+%
+%
+%
+%
+%
+%
+% \StopEventually{\setlength\IndexMin{200pt}  \PrintIndex  }
+%
+%
+% \section{The Implementation}
+%
+%    \begin{macrocode}
+%<*kernel>
+%    \end{macrocode}
+%
+% \subsection{File declaration}
+%    \begin{macrocode}
+\ProvidesFile{latex-lab-footnotes.ltx}
+        [2022-03-08 v0.5a changes to the footnote interfaces]
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+
+% latex.ltx
+
+        % not looked at yet
+
+\long\def\@mpfootnotetext#1{%
+  \global\setbox\@mpfootins\vbox{%
+    \unvbox\@mpfootins
+    \reset@font\footnotesize
+    \hsize\columnwidth
+    \@parboxrestore
+    \def\@currentcounter{mpfootnote}%
+    \protected@edef\@currentlabel
+         {\csname p@mpfootnote\endcsname\@thefnmark}%
+    \color@begingroup
+      \@makefntext{%
+        \rule\z@\footnotesep\ignorespaces#1\@finalstrut\strutbox}%
+    \par
+    \color@endgroup}}
+
+
+\def\@makefnmark{\hbox{\@textsuperscript{\normalfont\@thefnmark}}}
+
+
+
+\def\@mpfn{footnote}
+\def\thempfn{\thefootnote}
+
+
+
+%-------------------------------------
+
+
+\ExplSyntaxOn
+
+
+\cs_new:Npn \fn_step_fnmark:nn #1#2 {
+  \tl_if_novalue:nTF {#1}
+    {
+      \stepcounter {#2}
+      \protected@xdef \@thefnmark { \cs:w the#2 \cs_end: }
+    }
+    {
+     \group_begin:
+        \int_set:cn { c@#2 }{ #1 }
+        \unrestored@protected@xdef \@thefnmark { \cs:w the#2 \cs_end: }
+     \group_end:
+    }
+}
+
+\cs_new:Npn \fn_set_fnmark:nn #1#2 {
+  \tl_if_novalue:nTF {#1}
+    {
+      \protected@xdef \@thefnmark { \cs:w the#2 \cs_end: }
+    }
+    {
+     \group_begin:
+        \int_set:cn { c@#2 }{ #1 }
+        \unrestored@protected@xdef \@thefnmark { \cs:w the#2 \cs_end: }
+     \group_end:
+    }
+}
+
+%-------------------------------------
+
+% the debugging code is just temp
+\bool_new:N \g_fnote_debug_bool
```

Comment:
These are `fnote` but above we have `fn`: what's the reasoning?

Reply:
do we? different times of development ...  not intentional.  I now normalized to fnote as prefix for now